src/routes/Tool/Competences.tsx

```diff
@@ -146,6 +92,30 @@ const Competences = () => {
   const [keyFigure, setKeyFigure] = React.useState('a');
   const [order, setOrder] = React.useState('descending');
 
+  const [debouncedTyotehtava, tyotehtava, setTyotehtava] = useDebounceState('', 500);
+
+  const methods = useForm<ToolsCompetencesForm>({
+    mode: 'onChange',
+    resolver: zodResolver(
+      z.object({
+        osaamiset: z.array(
+          z.object({
+            id: z.string().min(1),
+          }),
+        ),
+      }),
+    ),
+    defaultValues: async () => {
+      return Promise.resolve({
+        osaamiset: [],
+      });
+    },
+  });
+
+  const { isLoading } = useFormState({
+    control: methods.control,
+  });
```

Comment:
Is React Form Hook a little redundant at this part?

Reply:
I thought that when submitting the selected competences later down the road it is done with form's `onSubmit`

Reply:
This should be now changed to a lot simpler.